src/prop13/graphlibrary/Edge.java

```diff
@@ -5,13 +5,33 @@
  */
 public abstract class Edge
 {
-    protected Node n1;
-    protected Node n2;
-    protected int weight;
+    private Node n1;
+    private Node n2;
+    private int weight;
 
-    public Edge()
-    {
-    }
+	public Node getN1() {
```

Comment:
I didn't implement getters for n1 and n2 (and used that "weird" getNode and getNeighbor) to try to make both nodes have the same status on the edge (no origin or destiny, you can only get one the nodes of the edge in an arbitrary way, the other you have to get with getNeighbor) due to the fact that we will be working with undirected edges.